Can't install on windows

Tries to compile some c files and fails. Thought it was pure python?!

```
> c:\Python34\Scripts\virtualenv.exe venv-3.4
> venv-3.4\Scripts\activate
> pip install thriftpy
```

```
Collecting thriftpy
  Downloading thriftpy-0.2.0.tar.gz (177kB)
    100% |################################| 180kB 911kB/s
Collecting ply==3.4 (from thriftpy)
  Downloading ply-3.4.tar.gz (138kB)
    100% |################################| 139kB 1.2MB/s
Installing collected packages: ply, thriftpy
  Running setup.py install for ply
  Running setup.py install for thriftpy
    building 'thriftpy.transport.cybase' extension
    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python34\include -IC:\Python34\include /Tcthriftpy/transport/cybase.c /Fobuild\temp.win3
2-3.4\Release\thriftpy/transport/cybase.obj
    cybase.c
    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:C:\Python34\Libs /LIBPATH:D:\pcs\trunk\venv-3.4\libs /LIBPATH:D:\pcs\trunk\venv-3.4\PCbuil
d /EXPORT:PyInit_cybase build\temp.win32-3.4\Release\thriftpy/transport/cybase.obj /OUT:build\lib.win32-3.4\thriftpy\transport\cybase.pyd /IMPLIB:build\temp.win32-3.4\Release\thriftpy/transport\cybase
.lib /MANIFESTFILE:build\temp.win32-3.4\Release\thriftpy/transport\cybase.pyd.manifest
       Creating library build\temp.win32-3.4\Release\thriftpy/transport\cybase.lib and object build\temp.win32-3.4\Release\thriftpy/transport\cybase.exp
    building 'thriftpy.transport.buffered.cybuffered' extension
    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python34\include -IC:\Python34\include /Tcthriftpy/transport/buffered/cybuffered.c /Fobu
ild\temp.win32-3.4\Release\thriftpy/transport/buffered/cybuffered.obj
    cybuffered.c
    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:C:\Python34\Libs /LIBPATH:D:\pcs\trunk\venv-3.4\libs /LIBPATH:D:\pcs\trunk\venv-3.4\PCbuil
d /EXPORT:PyInit_cybuffered build\temp.win32-3.4\Release\thriftpy/transport/buffered/cybuffered.obj /OUT:build\lib.win32-3.4\thriftpy\transport\buffered\cybuffered.pyd /IMPLIB:build\temp.win32-3.4\Rel
ease\thriftpy/transport/buffered\cybuffered.lib /MANIFESTFILE:build\temp.win32-3.4\Release\thriftpy/transport/buffered\cybuffered.pyd.manifest
       Creating library build\temp.win32-3.4\Release\thriftpy/transport/buffered\cybuffered.lib and object build\temp.win32-3.4\Release\thriftpy/transport/buffered\cybuffered.exp
    building 'thriftpy.transport.memory.cymemory' extension
    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python34\include -IC:\Python34\include /Tcthriftpy/transport/memory/cymemory.c /Fobuild\
temp.win32-3.4\Release\thriftpy/transport/memory/cymemory.obj
    cymemory.c
    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:C:\Python34\Libs /LIBPATH:D:\pcs\trunk\venv-3.4\libs /LIBPATH:D:\pcs\trunk\venv-3.4\PCbuil
d /EXPORT:PyInit_cymemory build\temp.win32-3.4\Release\thriftpy/transport/memory/cymemory.obj /OUT:build\lib.win32-3.4\thriftpy\transport\memory\cymemory.pyd /IMPLIB:build\temp.win32-3.4\Release\thrif
tpy/transport/memory\cymemory.lib /MANIFESTFILE:build\temp.win32-3.4\Release\thriftpy/transport/memory\cymemory.pyd.manifest
       Creating library build\temp.win32-3.4\Release\thriftpy/transport/memory\cymemory.lib and object build\temp.win32-3.4\Release\thriftpy/transport/memory\cymemory.exp
    building 'thriftpy.transport.framed.cyframed' extension
    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python34\include -IC:\Python34\include /Tcthriftpy/transport/framed/cyframed.c /Fobuild\
temp.win32-3.4\Release\thriftpy/transport/framed/cyframed.obj
    cyframed.c
    c:\users\sesa334283\appdata\local\temp\pip-build-saq5mh71\thriftpy\thriftpy\transport\framed\../../protocol/cybin/endian_port.h(34) : fatal error C1083: Cannot open include file: 'endian.h': No su
ch file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\VC\\BIN\\cl.exe' failed with exit status 2
    Complete output from command D:\pcs\trunk\venv-3.4\Scripts\python.exe -c "import setuptools, tokenize;__file__='C:\\Users\\SESA33~1\\AppData\\Local\\Temp\\pip-build-saq5mh71\\thriftpy\\setup.py';e
xec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\SESA33~1\AppData\Local\Temp\pip-w3af_y5n-record\install-record.txt --s
ingle-version-externally-managed --compile --install-headers D:\pcs\trunk\venv-3.4\include\site\python3.4:
    running install

    running build

    running build_py

    creating build

    creating build\lib.win32-3.4

    creating build\lib.win32-3.4\thriftpy

    copying thriftpy\hook.py -> build\lib.win32-3.4\thriftpy

    copying thriftpy\rpc.py -> build\lib.win32-3.4\thriftpy

    copying thriftpy\server.py -> build\lib.win32-3.4\thriftpy

    copying thriftpy\thrift.py -> build\lib.win32-3.4\thriftpy

    copying thriftpy\tornado.py -> build\lib.win32-3.4\thriftpy

    copying thriftpy\utils.py -> build\lib.win32-3.4\thriftpy

    copying thriftpy\_compat.py -> build\lib.win32-3.4\thriftpy

    copying thriftpy\__init__.py -> build\lib.win32-3.4\thriftpy

    creating build\lib.win32-3.4\thriftpy\parser

    copying thriftpy\parser\exc.py -> build\lib.win32-3.4\thriftpy\parser

    copying thriftpy\parser\lexer.py -> build\lib.win32-3.4\thriftpy\parser

    copying thriftpy\parser\parser.py -> build\lib.win32-3.4\thriftpy\parser

    copying thriftpy\parser\__init__.py -> build\lib.win32-3.4\thriftpy\parser

    creating build\lib.win32-3.4\thriftpy\protocol

    copying thriftpy\protocol\binary.py -> build\lib.win32-3.4\thriftpy\protocol

    copying thriftpy\protocol\exc.py -> build\lib.win32-3.4\thriftpy\protocol

    copying thriftpy\protocol\json.py -> build\lib.win32-3.4\thriftpy\protocol

    copying thriftpy\protocol\__init__.py -> build\lib.win32-3.4\thriftpy\protocol

    creating build\lib.win32-3.4\thriftpy\transport

    copying thriftpy\transport\socket.py -> build\lib.win32-3.4\thriftpy\transport

    copying thriftpy\transport\__init__.py -> build\lib.win32-3.4\thriftpy\transport

    creating build\lib.win32-3.4\thriftpy\transport\buffered

    copying thriftpy\transport\buffered\__init__.py -> build\lib.win32-3.4\thriftpy\transport\buffered

    creating build\lib.win32-3.4\thriftpy\transport\framed

    copying thriftpy\transport\framed\__init__.py -> build\lib.win32-3.4\thriftpy\transport\framed

    creating build\lib.win32-3.4\thriftpy\transport\memory

    copying thriftpy\transport\memory\__init__.py -> build\lib.win32-3.4\thriftpy\transport\memory

    running build_ext

    building 'thriftpy.transport.cybase' extension

    creating build\temp.win32-3.4

    creating build\temp.win32-3.4\Release

    creating build\temp.win32-3.4\Release\thriftpy

    creating build\temp.win32-3.4\Release\thriftpy\transport

    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python34\include -IC:\Python34\include /Tcthriftpy/transport/cybase.c /Fobuild\temp.win3
2-3.4\Release\thriftpy/transport/cybase.obj

    cybase.c

    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:C:\Python34\Libs /LIBPATH:D:\pcs\trunk\venv-3.4\libs /LIBPATH:D:\pcs\trunk\venv-3.4\PCbuil
d /EXPORT:PyInit_cybase build\temp.win32-3.4\Release\thriftpy/transport/cybase.obj /OUT:build\lib.win32-3.4\thriftpy\transport\cybase.pyd /IMPLIB:build\temp.win32-3.4\Release\thriftpy/transport\cybase
.lib /MANIFESTFILE:build\temp.win32-3.4\Release\thriftpy/transport\cybase.pyd.manifest

       Creating library build\temp.win32-3.4\Release\thriftpy/transport\cybase.lib and object build\temp.win32-3.4\Release\thriftpy/transport\cybase.exp

    building 'thriftpy.transport.buffered.cybuffered' extension

    creating build\temp.win32-3.4\Release\thriftpy\transport\buffered

    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python34\include -IC:\Python34\include /Tcthriftpy/transport/buffered/cybuffered.c /Fobu
ild\temp.win32-3.4\Release\thriftpy/transport/buffered/cybuffered.obj

    cybuffered.c

    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:C:\Python34\Libs /LIBPATH:D:\pcs\trunk\venv-3.4\libs /LIBPATH:D:\pcs\trunk\venv-3.4\PCbuil
d /EXPORT:PyInit_cybuffered build\temp.win32-3.4\Release\thriftpy/transport/buffered/cybuffered.obj /OUT:build\lib.win32-3.4\thriftpy\transport\buffered\cybuffered.pyd /IMPLIB:build\temp.win32-3.4\Rel
ease\thriftpy/transport/buffered\cybuffered.lib /MANIFESTFILE:build\temp.win32-3.4\Release\thriftpy/transport/buffered\cybuffered.pyd.manifest

       Creating library build\temp.win32-3.4\Release\thriftpy/transport/buffered\cybuffered.lib and object build\temp.win32-3.4\Release\thriftpy/transport/buffered\cybuffered.exp

    building 'thriftpy.transport.memory.cymemory' extension

    creating build\temp.win32-3.4\Release\thriftpy\transport\memory

    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python34\include -IC:\Python34\include /Tcthriftpy/transport/memory/cymemory.c /Fobuild\
temp.win32-3.4\Release\thriftpy/transport/memory/cymemory.obj

    cymemory.c

    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:C:\Python34\Libs /LIBPATH:D:\pcs\trunk\venv-3.4\libs /LIBPATH:D:\pcs\trunk\venv-3.4\PCbuil
d /EXPORT:PyInit_cymemory build\temp.win32-3.4\Release\thriftpy/transport/memory/cymemory.obj /OUT:build\lib.win32-3.4\thriftpy\transport\memory\cymemory.pyd /IMPLIB:build\temp.win32-3.4\Release\thrif
tpy/transport/memory\cymemory.lib /MANIFESTFILE:build\temp.win32-3.4\Release\thriftpy/transport/memory\cymemory.pyd.manifest

       Creating library build\temp.win32-3.4\Release\thriftpy/transport/memory\cymemory.lib and object build\temp.win32-3.4\Release\thriftpy/transport/memory\cymemory.exp

    building 'thriftpy.transport.framed.cyframed' extension

    creating build\temp.win32-3.4\Release\thriftpy\transport\framed

    C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\BIN\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\Python34\include -IC:\Python34\include /Tcthriftpy/transport/framed/cyframed.c /Fobuild\
temp.win32-3.4\Release\thriftpy/transport/framed/cyframed.obj

    cyframed.c

    c:\users\sesa334283\appdata\local\temp\pip-build-saq5mh71\thriftpy\thriftpy\transport\framed\../../protocol/cybin/endian_port.h(34) : fatal error C1083: Cannot open include file: 'endian.h': No su
ch file or directory

    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\VC\\BIN\\cl.exe' failed with exit status 2

    ----------------------------------------
    Command "D:\pcs\trunk\venv-3.4\Scripts\python.exe -c "import setuptools, tokenize;__file__='C:\\Users\\SESA33~1\\AppData\\Local\\Temp\\pip-build-saq5mh71\\thriftpy\\setup.py';exec(compile(getattr(
tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\SESA33~1\AppData\Local\Temp\pip-w3af_y5n-record\install-record.txt --single-version-extern
ally-managed --compile --install-headers D:\pcs\trunk\venv-3.4\include\site\python3.4" failed with error code 1 in C:\Users\SESA33~1\AppData\Local\Temp\pip-build-saq5mh71\thriftpy
```

--- a/setup.py
+++ b/setup.py
@@ -3,6 +3,7 @@
 
 import re
 import sys
+import platform
 
 from os.path import join, dirname
 
@@ -42,9 +43,11 @@ ext_modules = []
 
 # pypy detection
 PYPY = "__pypy__" in sys.modules
+UNIX = platform.system() in ("Linux", "Darwin")
 
-# only build ext in CPython
-if not PYPY:
+# only build ext in CPython with UNIX platform
+if UNIX and not PYPY:
+    # rebuild .c files if cython available
     if CYTHON:
         cythonize("thriftpy/transport/cybase.pyx")
         cythonize("thriftpy/transport/**/*.pyx")

